clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -478,8 +478,12 @@ boolean joinGroupIfNeeded(final Timer timer) {
 
                 resetJoinGroupFuture();
                 synchronized (AbstractCoordinator.this) {
-                    rejoinReason = String.format("rebalance failed due to '%s' (%s)", exception.getMessage(), exception.getClass().getSimpleName());
-                    rejoinNeeded = true;
+                    final String shortReason = String.format("rebalance failed due to an exception: (%s)",
```

Comment:
nit: How about `rebalance failed due to %s`? `an exception` is redundant in my opinion.